java-frontend/src/test/files/se/LocksNotUnlockedCheck.java

```diff
@@ -324,4 +325,23 @@ public void lock() {
       }
     }
   }
+
+public class SonarFail {
+
+  private ConcurrentMap<String, ReentrantLock> locks = new ConcurrentHashMap<>();
+
+  public ReentrantLock getLock(String id) {
+    ReentrantLock lock = locks.computeIfAbsent(id, s -> new ReentrantLock());
+    lock.lock();
+    try {
+      if (locks.get(id) == lock) {
+        return lock;
+      }
+      return null;
+    } finally {
+      lock.unlock();
+    }
+  }
+
+}
```

Comment:
Same as previous comment, this can be much more simplified. We need only binary expression in the if statement to reproduce the problem